src/core/clipboard/clipboard.view.js

```diff
@@ -0,0 +1,176 @@
+        const action = model.action().shortcut;
+        const commands = this.commands;
+        action.register(selectionCommandManager, commands);
+
+        model.clipboardChanged.watch(e => {
+            const event = e.state.clipboardEvent;
+            this.onPaste(event);
+        })
+    }
+
+    onPaste(e) {
+        e.stopPropagation();
+        e.preventDefault();
+
+        const model = this.model;
+        const table = this.table;
+
+        const navigation = model.navigation();
+        let initialCell;
+        if (navigation.cell) {
+            initialCell = navigation.cell;
+        } else {
+            // further cell editing is based on navigation.cell, so selection unit should be specified as a cell
+            throw new AppError('clipboard.view', `For paste event switch selection unit to cell`);
+            return false;
+        }
+
+        const shortcut = { register: () => ({}) };
+        const editView = new EditCellView(model, table, shortcut);
+
+        const data = retrieveData(e);
+
+        let nextCellFlag = false;
+        let { rowIndex, columnIndex } = initialCell;
+
+        for (let i = 0, dataLength = data.length; i < dataLength; i++) {
+
+            const cells = data[i].split('\t');
+            for (let j = 0, cellsLength = cells.length; j < cellsLength; j++) {
+                const label = cells[j];
+                const isLast = j === cells.length - 1;
+                
+                if (initialCell && !nextCellFlag) {
+                    const cellView = table.body.cell(rowIndex, columnIndex).model();
+                    if (cellView) {
+                        editCell(cellView, editView, label);
+                    }
+                    if (!isLast) {
+                        nextCellFlag = true;
+                    }
+                } else if (nextCellFlag) {
+                    columnIndex += 1;
+                    const cellView = table.body.cell(rowIndex, columnIndex).model();
+                    if (cellView) {
+                        editCell(cellView, editView, label);
+                    }
+                }
+
+                if (isLast) {
+                    rowIndex = initialCell.rowIndex + (i + 1);
+                    columnIndex = initialCell.columnIndex;
+                    nextCellFlag = false;
+                }
+            }
+        }
+    }
+
+    get commands() {
+        const model = this.model;
+        const selectionState = model.selection();
+        const clipboardState = model.clipboard();
+        const { shortcut }  = clipboardState;
+
+        const commands = {
+            copy: new Command({
+                source: 'clipboard.view',
+                canExecute: () => selectionState.items.length > 0,
+                execute: () => {
+                    const selectionService = new SelectionService(model);
+                    const rowSelector = new RowSelector(model);
+                    const { items } = selectionState;
```

Comment:
try to make it like { items } = model.selection(), do not close model state please, there is no guarantee that state will be in a valid position